examples/reshare/src/self_channel.rs

```diff
@@ -0,0 +1,121 @@
+use bytes::Bytes;
+use commonware_cryptography::PublicKey;
+use commonware_p2p::{Receiver, Recipients, Sender};
+use futures::{
+    channel::mpsc::{self},
+    future::try_join,
+    select_biased, FutureExt as _, SinkExt, StreamExt as _,
+};
+use thiserror::Error;
+
+/// The possible errors that `self_channel` will produce.
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("network error: {0}")]
+    NetworkError(#[from] anyhow::Error),
+    #[error("the receiver has closed")]
+    ReceiverClosed,
+}
+
+/// A [Sender] allowing messages sent to yourself.
```

Comment:
I don't like this pattern (I think sending messages to yourself like this is error-prone/confusing and I excluded on purpose in `p2p`). If we want to allow this (would require a discussion), we should change `p2p` to support this rather than trying to workaround it here.